### DISCUSSION TITLE
About StreamPark's flink version support

### DISCUSSION THREAD

**Discussion by caicancai** [apache/streampark#3172] [Ideas]
StreamPark currently supports too many flink versions, many projects have rolled to support flink three versions, I think this measure is good, if you support too many flink versions, in order to be compatible with these flink versions at the same time we have to make some compromises at the code level, such as flink 1.14 and flink 1.15 There is a huge difference. And supporting too many Flink versions will affect upgrades between Java versions and Scala versions, such as Java to Java 11, Scala to Scala 2.3 or even 3.
So my idea is that following other projects and only supporting three versions of flink is enough

**Comment by caicancai** [comment]
@wolfboys @Al-assad @czy006

**Reply by czy006** [reply]
From my experience, I support this approach. We can refer to **Apache iceberg** and **Apache pamion**, both of which only support the 3 recent Spark or Flink versions.This can avoid developers from spending too much energy on compatibility maintenance of previous Flink versions.
